migration-engine/connectors/sql-migration-connector/src/flavour.rs

```diff
@@ -293,7 +297,7 @@ impl SqlFlavour for PostgresFlavour {
 
 /// Try to connect as an admin to a postgres database. We try to pick a default database from which
 /// we can create another database.
-async fn create_postgres_admin_conn(mut url: Url) -> ConnectorResult<(Quaint, DatabaseInfo)> {
+async fn create_postgres_admin_conn(url: &mut Url) -> ConnectorResult<(Quaint, DatabaseInfo)> {
```

Comment:
It's kind of confusing this API here. Why do we need to mutate the `Url` while connecting, why not just giving this a copy of it? Or change the interface so that we don't need to do a side-effect and mutate the input parameter in the same function. I know I know, test method, but this for me is bit of a code smell...

Reply:
It's because I wanted to reuse the schema-stripping code. I'll extract it to a separate function and do some more copying instead.